CRM/Financial/BAO/FinancialType.php

```diff
@@ -217,27 +217,33 @@ public static function permissionedFinancialTypes(&$permissions, $descriptions)
       return FALSE;
     }
     $financialTypes = CRM_Contribute_PseudoConstant::financialType();
-    $prefix = ts('CiviCRM') . ': ';
-    $actions = ['add', 'view', 'edit', 'delete'];
+    $actions = [
+      'add' => ts('add'),
+      'view' => ts('view'),
+      'edit' => ts('edit'),
+      'delete' => ts('delete'),
+    ];
+
     foreach ($financialTypes as $id => $type) {
-      foreach ($actions as $action) {
+      foreach ($actions as $action => $action_ts) {
         if ($descriptions) {
+          // FIXME-ts: this makes string translation assumptions.
           $permissions[$action . ' contributions of type ' . $type] = [
-            $prefix . ts($action . ' contributions of type ') . $type,
-            ts(ucfirst($action) . ' contributions of type ') . $type,
+            ts("CiviCRM: %1 contributions of type %2", [1 => $action_ts, 2 => $type]),
+            ts('%1 contributions of type %2', [1 => $action_ts, 2 => $type]),
```

Comment:
This removes the ucfirst, because there isn't an easy mb/utf8 function for ucfirst, and it would badly break non-lating languages.
Also, I'm still not happy with the end result, because it makes assumptions in how infinitives work in a language, but worst case translators can use "CiviCRM: '%1' contribution of type..", so that it's somewhat less awkward.